### PR TITLE
Migrate CI to GitHub Actions

### PR DESCRIPTION
**Pull request by WikiRik**
To allow more updates on this project with a possible maintenance reboot in the chaijs organization

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+jobs:
+    lint:
+        runs-on: ubuntu-latest
+
+        steps:
+            - uses: actions/checkout@v4
+            - uses: actions/setup-node@v4
+              with:
+                  node-version: latest
```
Comment (43081j) [inline .github/workflows/ci.yml:20]: can we use a matrix here? so we can use 18, 20 and latest like chai

Comment (WikiRik) [reply .github/workflows/ci.yml:20]: Yes, I started with a copy from chai's file but I wanted to minimize the changes as much as possible. So this currently follows what would have been done in Travis. 
I want to do additional PRs after this one for things like; updating dependencies to more recent versions, dropping support for unsupported Node versions and adding chai v5 to the peer dependencies and CI.

Comment (43081j) [reply .github/workflows/ci.yml:20]: take note of the small flood of PRs i just dropped FYI, so we don't duplicate work (though i'm done now i think)

& makes sense, feel free to leave this one as is and we can just bump it in a separate PR

**Comment by 43081j** [comment]
looks good to me

once we get the repo moved to the chai org, i should be able to merge

**Comment by WikiRik** [comment]
@43081j I see you have published 7.1.2, what do you want me to do with this PR? I can add chai 5 to the tests here, or remove older Node versions. You can also close this one if you want to combine it with any of your other PRs. I assume the next version of chai-as-promised that will come out is 8.0.0, right? Where the package is updated and migrated to ESM

**Comment by 43081j** [comment]
the next one will be 8.0.0

in the end i thought we may as well just bump the peer by itself with no other changes and we can focus on sorting 8.x separately then

if you can update the versions in this PR, ill merge it before the others

**Pull request merged**